application/lib/view/screens/home_screen.dart

```diff
@@ -1,29 +1,184 @@
-import 'package:application/utils/roots.dart';
+import 'package:application/utils/colors.dart';
+import 'package:application/utils/image_src.dart';
+import 'package:application/utils/styles/styles.dart';
 import 'package:application/utils/translation_en.dart';
+import 'package:application/view/widgets/auth_image_widget.dart';
+import 'package:application/view/widgets/horizontal_list.dart';
+import 'package:application/view/widgets/savings_card.dart';
+import 'package:application/view_model/home_screen_view_model.dart';
 import 'package:firebase_auth/firebase_auth.dart';
 import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
 
-class HomeScreen extends StatelessWidget {
+class HomeScreen extends StatefulWidget {
   const HomeScreen({super.key});
 
+  @override
+  State<HomeScreen> createState() => _HomeScreenState();
+}
+
+class _HomeScreenState extends State<HomeScreen> {
+  @override
+  void initState() {
+    super.initState();
+    User? user = FirebaseAuth.instance.currentUser;
+
+    String userId = user!.uid;
+    Future.microtask(() =>
+        Provider.of<HomeScreenViewModel>(context, listen: false)
+            .fetchData(userId));
+  }
+
   @override
   Widget build(BuildContext context) {
+    final homeScreenViewModel = Provider.of<HomeScreenViewModel>(context);
+
     return Scaffold(
-      appBar: AppBar(
-        title: const Text(TranslationEN.home),
-        actions: [
-          IconButton(
-            onPressed: () {
-              FirebaseAuth.instance.signOut();
-              Navigator.of(context).pushNamedAndRemoveUntil(
-                  Roots.startScreen, (Route<dynamic> route) => false);
-            },
-            icon: const Icon(Icons.exit_to_app),
+      backgroundColor: AppColor.lightBackgroundColor,
+      body: SafeArea(
+        child: SingleChildScrollView(
+          child: Padding(
+            padding: const EdgeInsets.all(15),
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.center,
+              mainAxisSize: MainAxisSize.min,
+              children: [
+                Align(
+                  alignment: Alignment.centerLeft,
+                  child: Text(
+                    '${TranslationEN.greet}, ${homeScreenViewModel.userFirstName}',
+                    style: TextStyle(
+                      fontWeight: FontWeight.bold,
+                      fontSize: MediaQuery.of(context).textScaleFactor * 30,
```

Comment:
Abban az esetben, ha mindenképp szükség van a MediaQuery-re, akkor ne az összes attributumra iratkozzunk fel a MediaQuery.of() függvénnyel, mert így minden változásra frissülni fog a képernyő, még arra is ha fényerőt állít a user. Ilyenkor használjuk konkrétan azt, amire szükségünk van, ebben az esetben MediaQuery.textScaleFactorOf()